core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -2248,7 +2250,7 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
     // original result task 1.0 succeed
     runEvent(makeCompletionEvent(taskSets(1).tasks(1), Success, 42))
     sc.listenerBus.waitUntilEmpty()
-    assert(completedStage === List(0, 1, 1, 0))
+    assert(completedStage === List(0, 1, 1, 0, 1))
```

Comment:
The last one added is the resubmit stage (FetchFailed) and in waiting stages. We will kill it and one more SparkListenerStageCompleted event will be added ( see markStageAsFinished)

Reply:
After changing to judging by `failedAttemptIds`,  it won't  cancel. Because all tasks finished already , and in `markStageAsFinished` will remove `failedAttemptIds` if no error message.

Reply:
@yabola I'm curious about the difference here. With the current approach, doesn't the stage still has to be killed because of `failedAttemptIds.nonEmpty`?

Reply:
In addition, the result stage will definitely kill all tasks after success, please see here
https://github.com/apache/spark/blob/939129ec01af7f7b6dbec737f2d4149d2fc0d9a3/core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala#L1960

Reply:
OK, sounds good to me.